cmd/operator-sdk/generate/csv.go

```diff
@@ -53,35 +55,58 @@ for the operator. This file is used to publish the operator to the OLM Catalog.
 
 A CSV semantic version is supplied via the --csv-version flag. If your operator
 has already generated a CSV manifest you want to use as a base, supply its
-version to --from-version. Otherwise the SDK will scaffold a new CSV manifest.
+version to --from-version. Otherwise the SDK will scaffold a new CSV manifest.`,
+		Example: `
+TODO
+	`,
 
-Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config.yaml`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			// The CSV generator assumes that the deploy and pkg directories are
-			// present at runtime, so this command must be run in a project's root.
-			projutil.MustInProjectRoot()
-
 			if len(args) != 0 {
 				return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
 			}
 			if err := c.validate(); err != nil {
 				return fmt.Errorf("error validating command flags: %v", err)
 			}
+			// Default for crd dir if unset
+			if c.crdDir == "" {
+				c.crdDir = c.deployDir
+			}
 			if err := c.run(); err != nil {
 				log.Fatal(err)
 			}
 			return nil
 		},
 	}
 
-	cmd.Flags().StringVar(&c.csvVersion, "csv-version", "", "Semantic version of the CSV")
+	cmd.Flags().StringVar(&c.csvVersion, "csv-version", "",
+		"Semantic version of the CSV")
 	if err := cmd.MarkFlagRequired("csv-version"); err != nil {
 		log.Fatalf("Failed to mark `csv-version` flag for `generate csv` subcommand as required: %v", err)
 	}
 	cmd.Flags().StringVar(&c.fromVersion, "from-version", "",
 		"Semantic version of an existing CSV to use as a base")
-	cmd.Flags().StringVar(&c.csvConfigPath, "csv-config", "",
-		"Path to CSV config file. Defaults to deploy/olm-catalog/csv-config.yaml")
+
+	cmd.Flags().StringVar(&c.deployDir, "deploy-dir", "deploy",
+		`Project relative path to root directory for operator manifests (Deployment, RBAC, CRDs).
+The CSV file contents will be generated from the manifests present in this directory. 
+`)
+	cmd.Flags().StringVar(&c.apisDir, "apis-dir", filepath.Join("pkg", "apis"),
+		`Project relative path to root directory for API type defintions.
+The CSV annotation comments will be parsed from the Go types under this path to
+fill out metadata for owned APIs in spec.customresourcedefinitions.owned.
+`)
+	cmd.Flags().StringVar(&c.crdDir, "crd-dir", "",
+		`Project relative path to root directory for for CRD manifests.
+Used when --update-crds is set to copy over CRD manifests to the CSV bundle directory.
+Note: The CSV generator only uses this to copy the CRD manifests.
+The CSV contents for spec.customresourcedefinitions.owned will still be updated
+from the CRD manifests in the deploy directory specified by --deploy-dir.
+If unset, it defaults to the same value as --deploy-dir.
```

Comment:
I think the above text is too verbose. Just `Project relative path to the root directory for CRD manifests` shows enough here. Could you w not add the rest in so example?